resources/analysisTools/snvPipeline/snv_extractor.pl

```diff
@@ -0,0 +1,217 @@
+
+my %genes;
+my %somatic_functional;
+my $whitelist_header;
+my @whitelist_header;
+my %whitecol;
+my @snv_genes;
+
+while(<IN>)
+{
+	chomp;
+	next if($_ =~ /^#/);
+	my @line = split("\t", $_);
+        if($line[$col{"CONFIDENCE"}] >= $whitelist_minconf && $line[$col{"CONFIDENCE"}] <= $whitelist_maxconf ){
+# Relaxed settings for genes in Whitelist
+
+             if($line[$col{"ANNOTATION_control"}] eq "somatic" && $line[$col{"ANNOVAR_FUNCTION"}] eq "exonic"){push(@{$genes{$line[$col{"GENE"}]}} ,$_);}
+
+        }
+
+	next if($line[$col{"CONFIDENCE"}] < $minconf);
+
+	if($line[$col{"ANNOTATION_control"}] eq "somatic" && $line[$col{"RECLASSIFICATION"}] !~ /lowCov_SNP_support_germline/){print SOM $_, "\n";}
+	
+	if($line[$col{"ANNOTATION_control"}] eq "somatic" && $line[$col{"RECLASSIFICATION"}] !~ /lowCov_SNP_support_germline/ && $line[$col{"ANNOVAR_FUNCTION"}] !~ /ncRNA/ && ($line[$col{"EXONIC_CLASSIFICATION"}] =~ /nonsynonymous/ || $line[$col{"EXONIC_CLASSIFICATION"}] =~ /stopgain/ ||$line[$col{"EXONIC_CLASSIFICATION"}] =~ /stoploss/ || $line[$col{"ANNOVAR_FUNCTION"}] =~ /splicing/)){
+		print COD $_, "\n";
+# "Normal somatic coding SNVs" For Whitelist
+                push(@{$somatic_functional{$line[$col{"GENE"}]}} ,$_);
+
+
+	}
+	if($extractNcRNA == 1 && $line[$col{"ANNOTATION_control"}] eq "somatic" && $line[$col{"RECLASSIFICATION"}] !~ /lowCov_SNP_support_germline/ && ($line[$col{"ANNOVAR_FUNCTION"}] =~ /ncRNA_exonic/ || $line[$col{"ANNOVAR_FUNCTION"}] =~ /ncRNA_splicing/)){
+		print NCRNA $_, "\n";
+	}
+	if($extractsyn == 1 && $line[$col{"ANNOTATION_control"}] eq "somatic" && $line[$col{"RECLASSIFICATION"}] !~ /lowCov_SNP_support_germline/ && ($line[$col{"EXONIC_CLASSIFICATION"}] =~ /^synonymous/ )){
+		print SYN $_, "\n";
+	}
+	
+	if($line[$col{"ANNOTATION_control"}] eq "germline" && $line[$col{"ANNOVAR_FUNCTION"}] !~ /ncRNA/ && ($line[$col{"EXONIC_CLASSIFICATION"}] =~ /nonsynonymous/ || $line[$col{"EXONIC_CLASSIFICATION"}] =~ /stopgain/ ||$line[$col{"EXONIC_CLASSIFICATION"}]  =~ /stoploss/ || $line[$col{"ANNOVAR_FUNCTION"}] =~ /splicing/)){
+		print GER $_, "\n";
+	}
+}
+
+close IN;
+
+if ($whitelist ne "NA"){
+     #whitelisted and "regular" somatic mutations
+     my $outsomwhite = $pid."_somatic_functional_snvs_conf_".$minconf."_to_10_plus_whitelisted_genes_conf_".$whitelist_minconf."_to_".$whitelist_maxconf.".vcf";
+    
+     open(WHITE, ">$outsomwhite") or die "Could not open the file $outsomwhite\n";
+
+     print WHITE $head, "\n";
+
+
+    if (open(WHITELIST, "<$whitelist")){	
+        while(<WHITELIST>)
+	{
+		chomp;
+		$whitelist_header=$_;
+		last if($_ =~ /^chr/);
+
+	}
+    }else{
+        die "Could not open the whitelistfile: $whitelist\n";
+    }
+
+    @whitelist_header=split("\t", $whitelist_header);
+
+
+    $i = 0;
+    while($i < @whitelist_header)
+    {	
+        if($whitelist_header[$i] eq "gene"){$whitecol{"gene"} = $i; print "whitelisted genes in column ", $i+1,"\n";}
```

Comment:
Avoid unnecessarily long lines like this one. Use spaces after `if` etc. Compare to this

```Perl
if ($whitelist_header[$i] eq "gene") {
  $whitecol{"gene"} = $i; 
  print("whitelisted genes in column ", $i+1,"\n");
}
```